### PR TITLE
*: don't allocate SessionIndexUsageCollector when indexUsageLease equals 0

### LINKED ISSUE #23856: IndexUsageCollector will consume memory much and mucher when its closed.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Relase & New the session quickly.

### 2. What did you expect to see? (Required)

The memory usage is smooth.

### 3. What did you see instead (Required)

Memory goes higher, and the `IndexUsageCollector` consumes a lot.

Lark20210406-141650

### 4. What is your TiDB version? (Required)

current master.

### PR DESCRIPTION
**Pull request by rebelice**
### What problem does this PR solve?

Issue Number: close #23856 

Problem Summary:

### What is changed and how it works?

When `index-usage-sync-lease` is 0, we don't dump index usage information to KV, but we also will allocate `SessionIndexUsageCollector`. The bad thing is that we only recycle unneeded `SessionIndexUsageCollector` when we dump index usage information. That means we have no GC when `index-usage-sync-lease`. This causes unexpected memory usage.

Proposal: [xxx](url) 

What's Changed:

We modified the behavior.
Before allocate `SessionIndexUsageCollector`, we must confirm that `index-usage-sync-lease` is not 0.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

- No release note

### CODE REVIEW

File: session/session_test.go [hunk 1]
```diff
@@ -4231,3 +4231,10 @@ func (s *testSessionSerialSuite) TestParseWithParams(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(sb.String(), Equals, "SELECT 3")
 }
+
+func (s *testSessionSerialSuite) TestNoIndexUsageCollector(c *C) {
+	do := s.dom
+	c.Assert(do.NeedCollectIndexUsage(), IsFalse)
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	c.Assert(tk.Se.ExistSessionIndexUsageCollector(), IsFalse)
+}
```
Comment (qw4990) [inline session/session_test.go:4240]: This test seems not complete.
I suggest that we test it in this way:
1. enable this feature first, process some queries and then check `ExistSessionIndexUsageCollector` is not `nil`;
2. disable this feature, process the same queries and then check `ExistSessionIndexUsageCollector` is `nil`;

File: session/session.go [hunk 2]
```diff
@@ -2092,7 +2097,9 @@ func CreateSessionWithOpt(store kv.Storage, opt *Opt) (Session, error) {
 	// which periodically updates stats using the collected data.
 	if do.StatsHandle() != nil && do.StatsUpdating() {
 		s.statsCollector = do.StatsHandle().NewSessionStatsCollector()
-		s.idxUsageCollector = do.StatsHandle().NewSessionIndexUsageCollector()
+		if do.NeedCollectIndexUsage() {
```
Comment (coocood) [inline session/session.go:2100]: We can simply `atomic.LoadInt64(&indexUsageSyncLease) > 0'.

**Review by qw4990 — APPROVED** [review]

**Pull request merged**